node/malus/src/variants/suggest_garbage_candidate.rs

```diff
@@ -88,131 +92,166 @@ where
-					"Fetched validation data."
+					"Received request to second candidate",
 				);
 
-				let malicious_available_data =
-					AvailableData { pov: Arc::new(pov.clone()), validation_data };
-
-				let pov_hash = pov.hash();
-				let erasure_root = {
-					let chunks =
-						erasure::obtain_chunks_v1(n_validators as usize, &malicious_available_data)
-							.unwrap();
-
-					let branches = erasure::branches(chunks.as_ref());
-					branches.root()
-				};
-
-				let (collator_id, collator_signature) = {
-					use polkadot_primitives::v2::CollatorPair;
-					use sp_core::crypto::Pair;
-
-					let collator_pair = CollatorPair::generate().0;
-					let signature_payload = polkadot_primitives::v2::collator_signature_payload(
-						&relay_parent,
-						&candidate.descriptor().para_id,
-						&validation_data_hash,
-						&pov_hash,
-						&validation_code_hash,
-					);
+				// Need to draw value from Bernoulli distribution with given probability of success defined by the Clap parameter.
+				// Note that clap parameter must be f64 since this is expected by the Bernoulli::new() function, hence it must be converted.
+				let distribution = Bernoulli::new(self.percentage / 100.0).expect("Invalid probability! Percentage cannot be < 0 or > 100.");
 
-					(collator_pair.public(), collator_pair.sign(&signature_payload))
-				};
-
-				let malicious_commitments =
-					create_fake_candidate_commitments(&malicious_available_data.validation_data);
-
-				let malicious_candidate = CandidateReceipt {
-					descriptor: CandidateDescriptor {
-						para_id: candidate.descriptor().para_id,
-						relay_parent,
-						collator: collator_id,
-						persisted_validation_data_hash: validation_data_hash,
-						pov_hash,
-						erasure_root,
-						signature: collator_signature,
-						para_head: malicious_commitments.head_data.hash(),
-						validation_code_hash,
-					},
-					commitments_hash: malicious_commitments.hash(),
-				};
-				let malicious_candidate_hash = malicious_candidate.hash();
+				// Draw a random value from the distribution, where T: bool, and probability of drawing a 'true' value is = to percentage parameter,
+				// using thread_rng as the source of randomness.
+				let true_or_false = distribution.sample(&mut rand::thread_rng());
 
 				gum::debug!(
 					target: MALUS,
-					candidate_hash = ?candidate.hash(),
-					?malicious_candidate_hash,
-					"Created malicious candidate"
+					"😈 Sampled value from Bernoulli distribution is: {:?}",
+					&true_or_false,
 				);
 
-				// Map malicious candidate to the original one. We need this mapping to send back the correct seconded statement
-				// to the collators.
-				self.inner
-					.lock()
-					.expect("bad lock")
-					.map
-					.insert(malicious_candidate_hash, candidate.hash());
+				// Manipulate the message if sampled value is true
+				if true_or_false == true {
+					gum::info!(target: MALUS, "😈 Intercepted CandidateBackingMessage::Second; crafting a malicious candidate",);
```

Comment:
I'd suggest to move this `info!` logging to the place where we know the malicious candidate hash, so we can include it into the log message (in fact, I have a similar patch in my local branch but I forgot about it).

Reply:
I will add another `info!` further down to highlight this.

Reply:
I think it is better to keep only one `info!` block per candidate suggested.